### PR TITLE
iOS binding and demo

### PR DESCRIPTION
**Pull request by ksyeo1010**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,6 +178,42 @@ Replace `${ACCESS_KEY}` with yours obtained from Picovoice Console and `${MODEL_
 [default model file](/lib/common/cheetah_params.pv) (or your custom one). Finally, when done be sure to release
 resources acquired using `pv_cheetah_delete(handle)`.
 
+### iOS
+
+The Cheetah iOS binding is available via [Cocoapods](https://cocoapods.org/pods/Cheetah-iOS). To import it into your iOS project, add the following line to your Podfile and run `pod install`:
```
Comment (kenarsa) [inline README.md:183]: `Cocoapods` >>> `CocoaPods`

File: README.md [hunk 2]
```diff
@@ -164,6 +178,42 @@ Replace `${ACCESS_KEY}` with yours obtained from Picovoice Console and `${MODEL_
...
+  return audioFrame;
+}
+
+while true {
+  do {
+    let partialTranscript, isEndpoint = try cheetah.process(getNetAudioFrame())
+    if isEndpoint {
+      let finalTranscript = try cheetah.flush()
+    }
+  } catch { }
```
Comment (kenarsa) [inline README.md:211]: should we set the type to CheetahError?

File: README.md [hunk 3]
```diff
@@ -164,6 +178,42 @@ Replace `${ACCESS_KEY}` with yours obtained from Picovoice Console and `${MODEL_
...
+}
+```
+
+Replace `${ACCESS_KEY}` with yours obtained from Picovoice Console and `${MODEL_FILE}` with the name of the Cheetah model file name.
```
Comment (kenarsa) [inline README.md:215]: default or custom trained from Console (something like that and maybe a link to default model)

File: binding/ios/Cheetah.swift [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+//  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+//  file accompanying this source.
+//  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+//  specific language governing permissions and limitations under the License.
+//
+
+import PvCheetah
+
+/// iOS binding for Cheetah wake word engine. Provides a Swift interface to the Cheetah library.
```
Comment (kenarsa) [inline binding/ios/Cheetah.swift:12]: wake word engine ??? :(((

File: binding/ios/Cheetah.swift [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+        if accessKey.count == 0 {
+            throw CheetahInvalidArgumentError("AccessKey is required for Cheetah initialization")
+        }
+
+        if !FileManager().fileExists(atPath: modelPath) {
+            throw CheetahInvalidArgumentError("Model file at does not exist at '\(modelPath)'")
+        }
+
+        if endpointDuration <= 0 {
```
Comment (kenarsa) [inline binding/ios/Cheetah.swift:39]: I think `0` should be acceptable?

File: binding/ios/Cheetah.swift [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    ///      and be 16-bit linearly-encoded. Furthermore, Cheetah operates on single-channel audio.
+    /// - Throws: CheetahError
+    /// - Returns: Tuple of any newly-transcribed speech (if none is available then an empty string is returned) and a
+    ///   flag indicating if an endpoint has been detected.
+    public func process(_ pcm:[Int16]) throws -> (String, Bool) {
+        if handle == nil {
+            throw CheetahInvalidStateError("Cheetah must be initialized before processing")
+        }
+
+        if pcm.count == 0 {
```
Comment (kenarsa) [inline binding/ios/Cheetah.swift:92]: we should check that the length is equal to our expected frame length

File: binding/ios/README.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+- Private, All voice processing runs locally.
+- Accurate [[1]](https://github.com/Picovoice/speech-to-text-benchmark#results)
+- Compact and Computationally-Efficient [[1]](https://github.com/Picovoice/speech-to-text-benchmark#results)
+- Cross-Platform:
+    - Linux (x86_64)
+    - macOS (x86_64, arm64)
+    - Windows (x86_64)
+    - Raspberry Pi (4, 3)
+    - NVIDIA Jetson Nano
+    - Android and iOS
```
Comment (kenarsa) [inline binding/ios/README.md:18]: same order as the main README.md

File: binding/ios/README.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+## Installation
+
+The Cheetah iOS binding is available via [Cocoapods](https://cocoapods.org/pods/Cheetah-iOS). To import it into your iOS project, add the following line to your Podfile:
```
Comment (kenarsa) [inline binding/ios/README.md:22]: `Cocoapods` >>> `CocoaPods`

File: binding/ios/README.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    if isEndpoint {
+      let finalTranscript = try cheetah.flush()
+    }
+  } catch { }
+}
+
+```
+
+
+Replace `${ACCESS_KEY}` with yours obtained from [Picovoice Console]((https://console.picovoice.ai/)) and `${MODEL_FILE}` with the name of the Cheetah model file name. Finally, when done be sure to explicitly release the resources using `cheetah.delete()`.
```
Comment (kenarsa) [inline binding/ios/README.md:72]: a couple of lines where they can get the default and how to train a custom one

**Review by kenarsa — CHANGES_REQUESTED** [review]

**Comment by wasauce** [comment]
@ksyeo1010 does cheetah do punctuation and capitalization?

**Review by kenarsa — CHANGES_REQUESTED** [review]

**Pull request merged**